packages/core/src/lib/GooglePlay.ts

```diff
@@ -145,23 +161,75 @@ export class GooglePlay {
    * both ChromeOS and Android Releases.
    * @param packageName - The packageName of the versionCode we are looking up.
    */
-  async getLargestVersionCode(packageName: string): Promise<number> {
-    const edit = await this._googlePlayApi.edits.insert({packageName: packageName});
-    const editId = edit.data.id;
-    if (!editId) {
-      throw new Error('Could not create a Google Play edit');
-    }
+  async getLargestVersionCode(packageName: string, editId: string):
+      Promise<PlayOperationResult> {
     const bundleResponse =
-      await this._googlePlayApi.edits.bundles.list({packageName: packageName, editId: editId});
+        await this._googlePlayApi.edits.bundles.list({packageName: packageName, editId: editId});
     if (!bundleResponse.data.bundles) {
       throw new Error('No bundles found from Google Play');
     }
     const versionCode = Math.max(
         ...bundleResponse.data.bundles.map((bundle) => bundle.versionCode!!));
-    // cleanup
+
+    const playOpsResult: PlayOperationResult = {getLargestVersionCodeResult: versionCode};
+    return playOpsResult;
+  }
+
+  /**
+   * Starts an edit on the Play servers. This is the basis for any play publishing api operation.
+   * @param packageName - the packageName of the app we want to interact with.
+   *
+   */
+  private async startPlayOperation(packageName: string): Promise<string> {
+    const edit = await this._googlePlayApi.edits.insert({packageName: packageName});
+    const editId = edit.data.id;
+    if (!editId) {
+      throw new Error('Could not create a Google Play edit');
+    }
+    return editId;
+  }
+
+  /**
+   * Cancels the edit in progress on the Play server.
+   * @param packageName - The packageName of the app we are interacting with.
+   * @param editId - The editId that is currently in progress.
+   */
+  private async endPlayOperation(packageName: string, editId: string): Promise<void> {
     await this._googlePlayApi.edits.delete({editId: editId, packageName: packageName});
+  }
+
+  /**
+   * Checks to see if the version that we want to retain already exists within the Play Store.
+   * @param packageName - The packageName of the versionCode we are looking up.
+   * @param versionCode - The version code of the APK / Bundle we want to retain.
+   *
+   */
+  async versionExists(packageName: string, versionCode: number, editId: string):
+      Promise<PlayOperationResult> {
+    const playOpsResult: PlayOperationResult = {};
+
+    const uploadedApks =
+        await this._googlePlayApi.edits.apks.list({packageName: packageName, editId: editId});
+
+    let found = uploadedApks.data.apks?.find(async (obj) => obj.versionCode == versionCode);
```

Comment:
What does the `async` here do?

Reply:
I don't believe it actually does anything. Everything is loaded in from line 209/210. I will remove.